packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -78,6 +79,81 @@ void main() {
 
     const Key dragTarget = Key('orange box');
 
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: CustomScrollView(
+            key: dragTarget,
+            primary: true,
+            slivers: <Widget>[
+              SliverPersistentHeader(
+                floating: true,
+                pinned: true,
+                delegate: TestDelegate(settings: customSettings),
+              ),
+              SliverToBoxAdapter(
+                child: Container(
+                  height: 1200.0,
+                  color: Colors.orange[400],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox clipRect = tester.renderObject(find.byType(ClipRect).at(1));
+    final Transform transform = tester.firstWidget(
+      find.descendant(
+        of: find.byType(FlexibleSpaceBar),
+        matching: find.byType(Transform),
+      ),
+    );
+
+    // The current (200) is half way between the min (100) and max (300) and the
+    // lerp values used to calculate the scale are 1 and 1.5, so we check for 1.25.
+    expect(transform.transform.getMaxScaleOnAxis(), 1.25);
+
+    // The space bar rect always starts fully expanded.
+    expect(clipRect.size.height, maxExtent);
+
+    final Element actionTextBox = tester.element(find.text('title'));
+    final Text textWidget = actionTextBox.widget as Text;
+    final DefaultTextStyle defaultTextStyle = DefaultTextStyle.of(actionTextBox);
+
+    final TextStyle effectiveStyle = defaultTextStyle.style.merge(textWidget.style);
+    expect(effectiveStyle.color?.alpha, 128); // Which is alpha of .5
+
+    // We drag up to fully collapse the space bar.
+    await tester.drag(find.byKey(dragTarget), const Offset(0, -400.0));
+    await tester.pumpAndSettle();
+
+    expect(clipRect.size.height, minExtent);
+  });
+
+  testWidgets('Material2 - FlexibleSpaceBarSettings provides settings to a FlexibleSpaceBar', (WidgetTester tester) async {
```

Comment:
For the newly added test specifically for Material 3, can we also add `Material3 - ` in front of the test message?